.golangci.yaml

```diff
@@ -0,0 +1,19 @@
+run:
+  timeout: 1m
+  tests: true
+
+linters:
+  enable:
+    - asciicheck
+    - forcetypeassert
+    #- gocritic
+    - godot
+    - gofmt
+    - goimports
+    - misspell
+    #- revive
```

Comment:
I'm unsure about godot, but don't mind having it. It's actually a style that I prefer myself...

For reference, linting in Kubernetes uses:
https://github.com/kubernetes/kubernetes/blob/f70c26d49510adf0bed1736e12cd48849b33ed6c/hack/golangci.yaml#L29-L36

Let's add all of those.

Reply:
logcheck will not work as golangci-lint plugin and ginkgolinter is not useful here, so exclude those.

Reply:
Logcheck is not default, is it? https://golangci-lint.run/usage/linters/#ginkgolinter says defaults are: errcheck, gosimple, govet, ineffassign, staticcheck, typecheck, unused

Reply:
logcheck is a Kubernetes-specific plugin.

My proposal is to use:
```
linters:
  enable:
    - asciicheck
    - forcetypeassert
    - gocritic
    - godot
    - gofmt
    - goimports
    - misspell
    - stylecheck
```

That's a combination of what go-logr and Kubernetes enabled, with godot enabled because @byako suggested it.

Reply:
lgtm, most of revive complaints was about missing comment for exported entity, should be safe to skip it.

Reply:
Added style check, removed revive, added section from k/k linter config like @bart0sh suggested.

Reply:
My understanding was that @bart0sh only linked that additional part for reference and wasn't suggesting that we should do the same.

Kubernetes has to be very conservative for existing code. We can do better.

Reply:
/hold